src/NetscriptJSEvaluator.ts

```diff
@@ -76,15 +77,30 @@ function addDependencyInfo(script: Script, seenStack: Script[]) {
  * @param scripts array of other scripts on the server
  * @param seenStack A stack of scripts that were higher up in the import tree in a recursive call.
  */
-function generateLoadedModule(script: Script, scripts: Map<ScriptFilePath, Script>, seenStack: Script[]): LoadedModule {
+async function generateLoadedModule(
+  script: Script,
+  scripts: Map<ScriptFilePath, Script>,
+  seenStack: Script[],
+): Promise<LoadedModule> {
   // Early return for recursive calls where the script already has a URL
   if (script.mod) {
     addDependencyInfo(script, seenStack);
     return script.mod;
   }
 
+  const scriptJs = script.filename.endsWith(".js")
+    ? script.code
+    : //tsx can load js, ts, jsx and tsx.
```

Comment:
Although this is true, tsx is not a strict superset of ts and jsx. I can't find the reference right now, but there are cases where valid TS will be interpreted differently if it's compiled as TSX. Best to use the proper loader based on filetype.